dfaker.py

```diff
@@ -0,0 +1,387 @@
+from datetime import datetime, timedelta, tzinfo
+import time
+from pytz import timezone
+import pytz
+import uuid
+import json
+import argparse 
+import sys
+import cbg_equation
+import statsmodels.api as sm
+
+import random
+import numpy as np 
+
+params = {
+	'year' : 2015, #default datetime settings
+	'month' : 1,
+	'day' : 1,
+	'hour' : 0,
+	'minute' : 0,
+	'zone' : 'US/Pacific', #default zone
+	'num_days' : 30, #default days to generate data for
+	'file' : 'device-data.json', #default json file name 
+	'minify' : False, #compact storage option false by default 
+	'gaps' : False, #randomized gaps in data, off by default 
+	'smbg_freq' : 6 #default fingersticks per day
+}
+
+def parse(args, params):
+	if args.date:
```

Comment:
You should be able to make this more concise by parsing the dates with Python's `datetime.strptime`, which takes a format specification as described [here](https://docs.python.org/3.5/library/datetime.html#strftime-and-strptime-behavior). So the format YYYY-MM-DD is `%Y-%m-%d`. (These are pretty standard format specifications, even across programming languages, so a good thing to learn!)